### PR TITLE
Add forwards for topleft hud and clantag changes

### PR DESCRIPTION
**Pull request by kidfearless**
This adds 3 new forwards that will let people modify the topleft hud and clantags for individual clients.

### CODE REVIEW

File: addons/sourcemod/scripting/shavit-hud.sp [hunk 1]
```diff
@@ -1600,7 +1600,7 @@ void UpdateTopLeftHUD(int client, bool wait)
 
 			Call_Finish(result);
 			
-			if(result == Plugin_Continue || result == Plugin_Handled)
+			if(result == Plugin_Stop || result == Plugin_Handled)
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-hud.sp]: This seems to be the written the opposite way as the rest of the timer.
```SourcePawn
if(result != Plugin_Continue && result != Plugin_Changed)
{
    return;
}
```

File: addons/sourcemod/scripting/include/shavit.inc [hunk 2]
```diff
@@ -575,6 +575,34 @@ forward void Shavit_OnReplayStart(int client);
...
 
+/**
+ * Called when top left HUD updates.
+ *
+ * @param client					Client index that recieves the hud.
+ * @param target					Client index that recieves the hud.
+ * @param topleft					Reference to the HUD buffer.
+ * @return							Plugin_Handled or Plugin_Stop to block the HUD message from appearing. Anything else to pass along new values.
+ */
+forward Action Shavit_OnTopLeftHUD(int client, int target, char topleft[128]);
```
Comment (strafe) [inline addons/sourcemod/scripting/include/shavit.inc]: Generally, you would specify the buffer size in a separate parameter.
```sourcepawn
forward Action Shavit_OnTopLeftHUD(int client, int target, char[] topleft, int size);
```

File: addons/sourcemod/scripting/include/shavit.inc [hunk 3]
```diff
@@ -575,6 +575,34 @@ forward void Shavit_OnReplayStart(int client);
...
+
+/**
+ * Called before clan tag variables are processed
+ *
+ * @param client					Client index.
+ * @param clantag					Reference to the clan tag buffer.
+ * @return							Plugin_Handled or Plugin_Stop to block the clan tag from changing. Anything else to pass along new values.
+ */
+forward Action Shavit_OnClanTagChangePre(int client, char clantag[32]);
```
Comment (strafe) [inline addons/sourcemod/scripting/include/shavit.inc]: Buffer size again here.

File: addons/sourcemod/scripting/include/shavit.inc [hunk 4]
```diff
@@ -575,6 +575,34 @@ forward void Shavit_OnReplayStart(int client);
...
+
+/**
+ * Called after clan tag variables are processed
+ *
+ * @param client					Client index.
+ * @param customtag					Reference to the custom clan tag buffer.
+ * @return							Plugin_Handled or Plugin_Stop to block the clan tag from changing. Anything else to pass along new values.
+ */
+forward Action Shavit_OnClanTagChange(int client, char customtag[32]);
```
Comment (strafe) [inline addons/sourcemod/scripting/include/shavit.inc]: Buffer size again here.

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 5]
```diff
@@ -827,6 +835,17 @@ void UpdateClanTag(int client)
 	{
 		return;
 	}
+	Action result = Plugin_Continue;
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-misc.sp]: Could probably do with a newline to split this from the previous statement.

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 6]
```diff
@@ -827,6 +835,17 @@ void UpdateClanTag(int client)
...
+	Call_StartForward(gH_Forwards_OnClanTagChangePre);
+	Call_PushCell(client);
+	Call_PushStringEx(sTag, 32, SM_PARAM_STRING_COPY, SM_PARAM_COPYBACK);
+
+	Call_Finish(result);
+	
+	if(result == Plugin_Stop || result == Plugin_Handled)
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-misc.sp]: Written the opposite way again.

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 7]
```diff
@@ -888,6 +907,18 @@ void UpdateClanTag(int client)
...
 	ReplaceString(sCustomTag, 32, "{rank}", sRank);
 
+	result = Plugin_Continue;
+	Call_StartForward(gH_Forwards_OnClanTagChange);
+	Call_PushCell(client);
+	Call_PushStringEx(sCustomTag, 32, SM_PARAM_STRING_COPY, SM_PARAM_COPYBACK);
+
+	Call_Finish(result);
+	
+	if(result == Plugin_Stop || result == Plugin_Handled)
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-misc.sp]: Written the opposite way again.

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 8]
```diff
@@ -888,6 +907,18 @@ void UpdateClanTag(int client)
 	ReplaceString(sCustomTag, 32, "{tr}", sTrack);
 	ReplaceString(sCustomTag, 32, "{rank}", sRank);
 
+	result = Plugin_Continue;
```
Comment (shavitush) [inline addons/sourcemod/scripting/shavit-misc.sp]: `result` should be ignored completely as this is a post-forward

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 9]
```diff
@@ -827,6 +835,17 @@ void UpdateClanTag(int client)
 	{
 		return;
 	}
+	Action result = Plugin_Continue;
+	Call_StartForward(gH_Forwards_OnClanTagChangePre);
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-misc.sp]: This needs to be moved down to where `gH_Forwards_OnClanTagChange` is currently.

File: addons/sourcemod/scripting/shavit-misc.sp [hunk 10]
```diff
@@ -888,6 +907,18 @@ void UpdateClanTag(int client)
 	ReplaceString(sCustomTag, 32, "{tr}", sTrack);
 	ReplaceString(sCustomTag, 32, "{rank}", sRank);
 
+	result = Plugin_Continue;
+	Call_StartForward(gH_Forwards_OnClanTagChange);
```
Comment (strafe) [inline addons/sourcemod/scripting/shavit-misc.sp]: This needs to be put after `CS_SetClientClanTag` is called and changed to a post forward (void).

**Pull request merged**